cdm/core/src/main/java/ucar/nc2/iosp/IOServiceProvider.java

```diff
@@ -47,6 +47,16 @@ public interface IOServiceProvider {
    */
   boolean isValidFile(RandomAccessFile raf) throws IOException;
 
+  /**
+   * Check if this IOSP's isValidFile(raf) is a fast operation.
+   * If so, it will be checked earlier than the slower dynamically loaded IOSPs.
+   * For instance, file reads make isValidFile(raf) a slower check, especially for S3 files.
+   * @return true if isValidFile(raf) is a fast operation, else false.
+   */
+  default boolean isValidFileIsFast() {
```

Comment:
not sure about this name... instead of breaking the IOSPs into fast/slow groups, we could choose to break into "ones to check earlier" (regardless of speed) and "ones to check later". Then maybe call this `checkFirst`/`checkEarlier` or something